cmd/metrics-proxy/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/canonical/metrics-k8s-proxy/internal/handlers"
+	"github.com/canonical/metrics-k8s-proxy/internal/k8s"
+	"github.com/canonical/metrics-k8s-proxy/internal/util"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/gorilla/mux"
+)
+
+// Parses the label selector and timeout from command line arguments.
+func parseFlags() (map[string]string, time.Duration) {
+	labelSelector := flag.String("labels", "", "Label selector for watching pods (e.g., 'app=ztunnel')")
+	timeout := flag.Duration("timeout", 9*time.Second, "HTTP server read and write timeout (e.g., 15s, 1m)")
+	flag.Parse()
+
+	if *labelSelector == "" {
+		log.Fatal("Label selector is required (e.g., --labels app=ztunnel)")
+	}
+
+	return util.ParseLabels(*labelSelector), *timeout
+}
+
+// Initializes the Kubernetes client.
+func initK8sClient() kubernetes.Interface {
+	_, clientset, err := k8s.GetKubernetesClient(k8s.DefaultBuildConfigFunc, k8s.DefaultNewClientsetFunc)
+	if err != nil {
+		log.Fatalf("Error building Kubernetes config: %v", err)
+	}
+
+	return clientset
+}
+
+// Starts the HTTP server.
+func startServer(timeout time.Duration, pw *k8s.PodScrapeWatcher) *http.Server {
+	r := mux.NewRouter()
+
+	httpClient := &handlers.RealHTTPClient{Client: &http.Client{}}
+	metricsHandler := handlers.NewMetricsHandler(httpClient)
+
+	r.HandleFunc("/metrics", func(w http.ResponseWriter, r *http.Request) {
+		metricsHandler.ProxyMetrics(w, r, pw)
+	}).Methods(http.MethodGet)
+
+	server := &http.Server{
+		Handler:      r,
+		Addr:         "0.0.0.0:15090",
+		WriteTimeout: timeout,
+		ReadTimeout:  timeout,
```

Comment:
ended up leaving `ReadTimeoutHeader` cause of a linter warning 
`G112: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server`